### PR TITLE
Adds a watch flag to the get command

### PR DESCRIPTION
**Pull request by souleb**
Fix: #1586 

The new flag fetch and display the request ressource and then continue
watching the ressource until timeout or cancellation.

A single ressource/ressource type is supported.

Signed-off-by: Soule BA <soule@weave.works>

### CODE REVIEW

**Comment by souleb** [comment]
@stefanprodan this is still in progress, but would like your opinion. The package [cli-runtime](https://pkg.go.dev/k8s.io/cli-runtime) offers a set of helpers that would give an experience similar to kubectl.

**Comment by kingdonb** [comment]
I gave this a try, it works for Kustomizations so far, similar to kubectl the `all` resources alias is not supported with watch:

```
flux2 (flux-watch * u=)$ ./bin/flux get all -w
✗ expected a single resource type, but found all
yebyen@moo:~/go/src/github.com/fluxcd/flux2 (flux-watch * u=)$ ./bin/flux get ks -w
NAME       	READY	MESSAGE                                                        	REVISION                                     	SUSPENDED
flux-system	True 	Applied revision: main/839679c85b6abdee94628d869349110a2a7e282f	main/839679c85b6abdee94628d869349110a2a7e282f	False
cert-manager	True	Applied revision: main/839679c85b6abdee94628d869349110a2a7e282f	main/839679c85b6abdee94628d869349110a2a7e282f	False
my-secrets	True	Applied revision: main/839679c85b6abdee94628d869349110a2a7e282f	main/839679c85b6abdee94628d869349110a2a7e282f	False
flux-system	Unknown	reconciliation in progress	main/839679c85b6abdee94628d869349110a2a7e282f	False
flux-system	True	Applied revision: main/839679c85b6abdee94628d869349110a2a7e282f	main/839679c85b6abdee94628d869349110a2a7e282f	False
```

I assume this is meant to work eventually, part of the WIP that you mentioned:

```
$ ./bin/flux get source git -w
✗ This object type is not supported
```

This is good progress!

File: cmd/flux/get.go [hunk 1]
```diff
@@ -141,28 +155,116 @@ func (get getCommand) run(cmd *cobra.Command, args []string) error {
...
+		objToPrint := e.Object
+		var sink summarisable
+		switch k := get.apiType.kind; k {
+		case "Kustomization":
+			o := objToPrint.(*kustomizev1.Kustomization)
+			sink = kustomizationListAdapter{&kustomizev1.KustomizationList{
+				Items: []kustomizev1.Kustomization{
+					*o,
+				}}}
+		case "Alerts":
```
Comment (stefanprodan) [inline cmd/flux/get.go:223]: Please don't hardcode the kinds, the kind names are already declared in the CLI e.g. `alertType.kind` or in each API package e.g. `notificationv1.AlertKind`

Comment (souleb) [reply cmd/flux/get.go:223]: yes I'll do the necessary cleaning .

**Comment by stefanprodan** [comment]
@souleb please rebase with main, solve the conflicts and force push.

**Comment by stefanprodan** [comment]
@souleb please run `make test` before pushing changes. You have to adapt the client used for testing to match your changes.

**Comment by souleb** [comment]
> @souleb please run `make test` before pushing changes. You have to adapt the client used for testing to match your changes.

sorry about that. Fixed

**Review by stefanprodan — COMMENTED** [review]
LGTM

Thanks @souleb 🏅

**Review by stefanprodan — APPROVED** [review]

**Pull request merged**